### ISSUE TITLE
BlazePose Full Body Python?

### ISSUE THREAD

**Issue by danj96dev** [google/mediapipe#1228]
Is there a way to run the full body version of BlazePose not on mobile. Or to use mlkit without a mobile device?
The released version only offers the upper body but I need to run the full body model on a web server.

**Comment by shabnm** [comment]
Yes, I too faced the same issue, blazepose is for UpperBodyPoseTracker(), and if you want to do full body, you have to use mlkit-pose detection "https://developers.google.com/ml-kit/vision/pose-detection". Mlkit pose detection is made specially to run on android/iOs, but the blazepose upper body can be executed on web.

**Comment by danj96dev** [comment]
> 
> 
> Yes, I too faced the same issue, blazepose is for UpperBodyPoseTracker(), and if you want to do full body, you have to use mlkit-pose detection "https://developers.google.com/ml-kit/vision/pose-detection". Mlkit pose detection is made specially to run on android/iOs, but the blazepose upper body can be executed on web.

I need full body on web, I believe there's a tflite model for the full body that could be run on Python for example but I'm not certain. Is there any other way to run blazenet without a mobile device?

**Comment by shabnm** [comment]
> > Yes, I too faced the same issue, blazepose is for UpperBodyPoseTracker(), and if you want to do full body, you have to use mlkit-pose detection "https://developers.google.com/ml-kit/vision/pose-detection". Mlkit pose detection is made specially to run on android/iOs, but the blazepose upper body can be executed on web.
> 
> I need full body on web, I believe there's a tflite model for the full body that could be run on Python for example but I'm not certain. Is there any other way to run blazenet without a mobile device?

I used posenet to extract keypoints earlier and ran it on web. I recently started working on ml-kit and blazepose. no luck with them, not yet

**Comment by PrinceP** [comment]
Use the following repo for the web version for full-body blaze pose

[https://github.com/terryky/tfjs_webgl_app/tree/master/blazepose_fullbody](https://github.com/terryky/tfjs_webgl_app/tree/master/blazepose_fullbody
)

Also, for the python version, we would have to create the runner code for these models 

[https://github.com/PINTO0309/PINTO_model_zoo/tree/master/058_BlazePose_Full_Keypoints](https://github.com/PINTO0309/PINTO_model_zoo/tree/master/058_BlazePose_Full_Keypoints)

**Comment by mgyong** [comment]
MediaPipe v0.8 - Python solutions APIs for Face Mesh, Hands and Upper body pose released! https://github.com/google/mediapipe/releases/tag/0.8.0 Try the colab https://google.github.io/mediapipe/#mediapipe-in-python

**Comment by ashutoshsoni891** [comment]
> Use the following repo for the web version for full-body blaze pose
> 
> https://github.com/terryky/tfjs_webgl_app/tree/master/blazepose_fullbody
> 
> Also, for the python version, we would have to create the runner code for these models
> 
> https://github.com/PINTO0309/PINTO_model_zoo/tree/master/058_BlazePose_Full_Keypoints

HI @PrinceP , I have create my inference code but the points are highly inaccurate . Could you pleaes help me out 


import os
import numpy as np
import tensorflow as tf
from tensorflow.keras.models import load_model
import cv2

def set_env():
    gpus = tf.config.experimental.list_physical_devices('GPU')
    if gpus:
        try:
            # Restrict TensorFlow to only use the fourth GPU
            tf.config.experimental.set_visible_devices(gpus[0], 'GPU')

            # Currently, memory growth needs to be the same across GPUs
            for gpu in gpus:
                tf.config.experimental.set_memory_growth(gpu, True)
            logical_gpus = tf.config.experimental.list_logical_devices('GPU')
            print(len(gpus), "Physical GPUs,", len(logical_gpus), "Logical GPUs")
        except RuntimeError as e:
            # Memory growth must be set before GPUs have been initialized
            print(e)



def return_model(filepath):

    model = load_model(filepath)
    model.compile()
    return model

def get_preds(model , image):
    img = np.expand_dims(image , axis = 0)
    preds = model.predict(img)
    return preds

def convert_preds_to_xy(preds):

    kpts = []
    temp = preds[2][0]
    for x,y in zip(temp[::4] , temp[1::4]):
        kpts.append((int(y),int(x)))
    return kpts 

def infer():
    cap = cv2.VideoCapture(0)

    model = return_model('PINTO_model_zoo/058_BlazePose_Full_Keypoints/01_Accurate/01_float32/full_pose_landmark_39kp.h5')
    model.summary()

    while cap.isOpened():
        okay , frame = cap.read()
        if not okay :
            print('Cant open webcam , please try again!')
            break

        
        inframe = frame.copy()
        inframe_resize = cv2.resize(inframe , (256 , 256))/255
        
        preds = get_preds(model , inframe_resize)
        kpts = convert_preds_to_xy(preds)

        for point in kpts:
            cv2.circle(inframe_resize , point , 2, (255,255,0) , 1)
        cv2.imshow('Inference' , inframe_resize)

        if cv2.waitKey(1) & 0xFF == ord('q'):
            break
    cap.release()
    cv2.destroyAllWindows()


if __name__ == '__main__':
    set_env()
    infer()

**Comment by PrinceP** [comment]
@ashutoshsoni891 Can you check out the holistic approach also -  https://github.com/google/mediapipe/blob/master/docs/solutions/holistic.md
